homeassistant/components/ohme/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""The ohme integration."""
+
+from dataclasses import dataclass
+import logging
+
+from ohme import OhmeApiClient
+
+from homeassistant import core
+from homeassistant.config_entries import ConfigEntry
+
+from .const import PLATFORMS
+from .coordinator import OhmeCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+type OhmeConfigEntry = ConfigEntry[OhmeRuntimeData]
+
+
+@dataclass
+class OhmeRuntimeData:
+    """Store volatile data."""
+
+    client: OhmeApiClient
+    coordinator: OhmeCoordinator
+
+
+async def async_setup_entry(hass: core.HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ohme from a config entry."""
+
+    client = OhmeApiClient(entry.data["email"], entry.data["password"])
+    await client.async_create_session()
+    await client.async_update_device_info()
```

Comment:
do these need to be error handled? I would move those to `_async_setup` (which is called as part of `async_config_entry_first_refresh`). Then we have all client related operations in the coordinator. You can then also move the `OhmeApiClient` init to the coordinator's init (you will have a `self.config_entry` available there after calling `super().init`)